websites/S/Stremio/presence.ts

```diff
@@ -3,172 +3,404 @@ const presence = new Presence({
-			}
-		} else if (hash.includes("addons")) {
-			search = document.querySelector(
-				"#addons > div.filter > form > div.addon-search-input > input"
-			);
-			if (search?.value) {
-				presenceData.details = "Searching addons for:";
-				presenceData.state = search.value;
-			} else {
-				presenceData.state = active ?? "All";
-				title = document.querySelector(
-					"[class='ng-scope selected']"
-				)?.textContent;
-
-				presenceData.buttons = [
-					{
-						label: "Browse Addons",
-						url: href,
-					},
-				];
-				presenceData.details = `Browsing ${title}`;
+		appVersion = getAppVersion(hostname);
+
+	if (!privacy && search) {
+		let searchInput: HTMLInputElement;
+
+		if (appVersion === AppVersion.V4)
+			searchInput = document.querySelector("#global-search-field");
+		else searchInput = document.querySelector("input[class*='search-input']");
+
+		const searchValue = searchInput?.value;
+
+		if (searchValue) {
+			presenceData.details = `Searching for ${searchValue}`;
+			presenceData.smallImageKey = Assets.Search;
+			presence.setActivity(presenceData);
+			return;
+		}
+	}
+
+	switch (appVersion) {
+		case AppVersion.V4:
+		case AppVersion.V5: {
+			const video = findVideo(presence);
+
+			if (privacy) {
+				presenceData.details = video !== null ? "Watching" : "Browsing";
+				break;
 			}
-		} else if (hash.includes("settings")) {
-			presenceData.details = `${
-				document.querySelector("[class='ng-scope ng-binding active']")
-					?.textContent ?? "General"
-			} settings`;
-		} else if (hash.includes("/discover/")) {
-			if (active) {
-				presenceData.buttons = [
-					{
-						label: "Browse",
-						url: href,
-					},
-				];
-				presenceData.details = `Browsing: ${active}`;
-			} else presenceData.state = "Browsing Movies";
-		} else if (hash.includes("/library")) {
-			if (active) presenceData.details = `${active} Library`;
-			else presenceData.details = "Library";
-
-			presenceData.buttons = [
-				{
-					label: "View Library",
-					url: href,
-				},
-			];
-		} else if (hash.includes("/calendar")) {
-			presenceData.buttons = [
-				{
-					label: "View Calendar",
-					url: href,
-				},
-			];
-			presenceData.details = "Calendar";
-		} else if (hash.includes("player")) {
-			if (video?.duration) {
-				timestamp = presence.getTimestampsfromMedia(video);
-				pauseCheck = video?.paused ?? true;
-			} else if (!video.duration && document.querySelector("#controlbar-top")) {
-				let split = document
-					.querySelector("#play-progress-text")
-					?.textContent.split("/");
-				if (split?.[0]) {
-					split = split.map(s => s.trim());
-					timestamp = presence.getTimestamps(
-						presence.timestampFromFormat(split[0]),
-						presence.timestampFromFormat(split[1])
-					);
+
+			switch (hash.replace("#/", "").split("/").shift().split("?").shift()) {
+				case "":
+					presenceData.details = "Board";
+					presenceData.buttons = [
+						{
+							label: "View board",
```

Comment:
Button labels should be capitalized like so:
```ts
label: "Example Text",
```
(capitalize first letter of each word)

Reply:
Done.